core/src/test/java/net/sourceforge/jnlp/util/UrlWhiteListUtilsTest.java

```diff
@@ -0,0 +1,151 @@
+General Public License cover the whole combination.
+
+As a special exception, the copyright holders of this library give you
+permission to link this library with independent modules to produce an
+executable, regardless of the license terms of these independent modules, and
+to copy and distribute the resulting executable under terms of your choice,
+provided that you also meet, for each linked independent module, the terms and
+conditions of the license of that module. An independent module is a module
+which is not derived from or based on this library. If you modify this library,
+you may extend this exception to your version of the library, but you are not
+obligated to do so. If you do not wish to do so, delete this exception
+statement from your version. */
+
+package net.sourceforge.jnlp.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+
+import static net.sourceforge.jnlp.util.UrlWhiteListUtils.isUrlInWhitelist;
+
+public class UrlWhiteListUtilsTest {
+
+    @Test
+    public void expandLegalWhitelistUrlString() {
+        Assert.assertEquals("http://subdomain.domain.com:8080", UrlWhiteListUtils.expandedWhiteListUrlString("http://subdomain.domain.com:8080"));
+
+        Assert.assertEquals("https://domain.com:443", UrlWhiteListUtils.expandedWhiteListUrlString("domain.com"));
+        Assert.assertEquals("https://*.domain.com:443", UrlWhiteListUtils.expandedWhiteListUrlString("*.domain.com"));
+        Assert.assertEquals("https://*.domain.com", UrlWhiteListUtils.expandedWhiteListUrlString("*.domain.com:*"));
+
+        Assert.assertEquals("http://subdomain.domain.com:80", UrlWhiteListUtils.expandedWhiteListUrlString("http://subdomain.domain.com"));
+        Assert.assertEquals("https://subdomain.domain.com:443", UrlWhiteListUtils.expandedWhiteListUrlString("https://subdomain.domain.com"));
+
+        Assert.assertEquals("http://subdomain.domain.com", UrlWhiteListUtils.expandedWhiteListUrlString("http://subdomain.domain.com:*"));
+        Assert.assertEquals("https://subdomain.domain.com", UrlWhiteListUtils.expandedWhiteListUrlString("https://subdomain.domain.com:*"));
+
+        Assert.assertEquals("http://*:80", UrlWhiteListUtils.expandedWhiteListUrlString("http://*:80"));
+        Assert.assertEquals("https://*:443", UrlWhiteListUtils.expandedWhiteListUrlString("https://*:443"));
+
+        Assert.assertEquals("http://*", UrlWhiteListUtils.expandedWhiteListUrlString("http://*:*"));
+        Assert.assertEquals("https://*", UrlWhiteListUtils.expandedWhiteListUrlString("https://*:*"));
+    }
+
+    @Test(expected = MalformedURLException.class)
+    public void expandIllegalWhitelistUrlString() throws MalformedURLException {
+        URL expURL = new URL(UrlWhiteListUtils.expandedWhiteListUrlString("https://subdomain.domain.com:1*"));
+    }
+
```

Comment:
it would be also interesting if the MalformedURLException that is thrown could give the reason in its message, we can use the texts as defined in JIRA specs: Error: Invalid port, Error: Invalid host, etc. see OWS-383 (UI displayed column)